### ISSUE TITLE
Transforming 3.2.7 Change of Content

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag21#544]
This issue thread has been created to track the work being done by myself, @DavidMacDonald and @steverep (with input from Matt King) to revisit the wording and scope of 3.2.7 Change of Content.

**Comment by mbgower** [comment]
My latest stab at wording and concepts

> **Change of Content**
Programmatic notification of changes in content are provided for auto-updating content and status information

> status information
>- dynamic content such as success, progress or warning notices, which result from either user action or application status changes, and which do not take focus

> Programmatic notification
>- a mechanism which provides notice of changed content to user agents and assistive technology. The notification can include the actual changed content.

**Comment by DavidMacDonald** [comment]
This is workable for me. 

Would add, "errors" to the definition of Status information and content added/changed/hidden.

* dynamic content such as notice of success, error, progress, warning, new, changed, or hidden content, which result from either user action or application status changes, and which do not take focus.

**Comment by mbgower** [comment]
I always think in a "such as" statement, less is more. "Dynamic content" is a pretty wide term -- I think anyone would understand that dynamic content can be 'new' or 'changed'. I think we can clarify that in the understanding doc.
I'm concerned about using the term "hidden content". There are a lot of meanings of "hidden", and I initially misunderstood what you meant, I think. I believe "removed" is a better descriptor for what you intended? But I also believe that we really don't need anything else.
I'm okay swapping out "warning" with "error". I debated which to use and thought that error notices are a form of a warning. Here is is with the swap, to see if that scans better.

> dynamic content such as success, progress or error notices, which result from either user action or application status changes, and which do not take focus 

You haven't commented on the definition of 'programmatic notification'. I want it to be better, but I think it at least conveys the sense now.

**Comment by steverep** [comment]
> Programmatic notification of changes in content are provided for auto-updating content and status information

The problem with this is that it does not delineate between the types of auto-updating information.  This is great for a chat widget, but terrible for an ad carousel, stock ticker, or timer.

> a mechanism which provides notice of changed content to user agents and assistive technology. The notification can include the actual changed content.

1. Couldn't I just interpret "notice" as changing the DOM?  I don't think aria-live when I read this.
2. What do you mean by "can include"?  Would it be okay if it didn't?

**Comment by mbgower** [comment]
> it does not delineate between the types of auto-updating information

Does it need to? Is that not something that can be handled in the Understanding document and techniques? In example, we would have a technique covering content that auto-updates on a regular and frequent basis, such as carousels, stock tickers and timers. That technique's mechanism could be the assignment of a role such as timer, which designates a region of the page as frequently updating.

Did you envision a term that excludes different kinds of auto-updating content?

**Comment by mbgower** [comment]
>What do you mean by "can include"? Would it be okay if it didn't?

I was incorporating a phrase introduced by David. "Can" indicates that something is optional. I believe he was trying to drive at the fact that the notification could include the text string. I'm still trying to figure out a way to cleanly divide the act of programmatic notification from the message that may accompany it. Still not there.As I posted (about 5 seconds before your reply!) I think we can improve this definition. :)
Say, what about if we say "indication" instead of notification. Then we can work notification into the definition. Something like 
> Programmatic notification of changes in content is provided for auto-updating content and status information
> Programmatic Indication
> -a mechanism which provides an indication of changed content to user agents and assistive technology. The indication can include notification of the actual changed content.

Is that any better?

**Comment by steverep** [comment]
> Does it need to? Is that not something that can be handled in the Understanding document and techniques? In example, we would have a technique covering content that auto-updates on a regular and frequent basis, such as carousels, stock tickers and timers. That technique's mechanism could be the assignment of a role such as timer, which designates a region of the page as frequently updating.

Yes, I think it absolutely needs to somehow.  Your normative text says that programmatic notification is required, so you cannot explain in Understanding that it's okay to take a pass on certain types of content.

> Did you envision a term that excludes different kinds of auto-updating content?

No, my initial thought was to take care of this by forcing a role as I proposed on the [wiki page](https://www.w3.org/WAI/GL/wiki/Live_Region_Identification).  When the role is properly identified semantically, the default `aria-*` properties are usually appropriate.  However, even this is flawed because the role's default properties can just be overridden and my language did not disallow this.  I don't have a good solution yet.

> Programmatic Indication-a mechanism which provides an indication of changed content to user agents and assistive technology. The indication can include notification of the actual changed content.
> Is that any better?

Maybe slightly, but I'm still leaning strongly against trying to create a definition for "programmatic notification/indication" because I think it will always lead to problems.  One thing I thing my proposal got right was to recognize that we are basically just asking for programmatic determination of live regions with appropriate markup/politeness.  We have a definition for the former to use; the hard part is stating the latter.  In plain language, we want the SC to say:

Auto-updating and status update content can be programmatically determined, and the importance and relevance is set appropriately for assistive technologies.

**Comment by mbgower** [comment]
Ha, I had a draft that was thinking of the same thing regarding determinability

> Changes in content are programmatically determinable for auto-updating content and status information

**Comment by mraccess77** [comment]
I'd also mention that we want to include new content -- not just content that has changed.

**Comment by mbgower** [comment]
> I'd also mention that we want to include new content -- not just content that has changed.

"Changes in content" should cover modified, added or removed content. They all represent changes in content. That will be clarified in the Understanding document.

**Comment by awkawk** [comment]
Are we talking about programmatic determination of changes in content now, or still on the notification of changes of content?

Changes in content are already covered by WCAG 2.0, so I may just be misunderstanding what people are pushing for.

**Comment by steverep** [comment]
> I think my wording slightly better captures the idea that it's the changes in content we want determinable. 

Well, playing devil's advocate, the "change" is always going to be programmatically determinable if I move the virtual focus to it.  So I see 2 possible approaches:

1. Target the regions themselves as being programmatically determinable (similar to 4.1.2 and 1.3.1).  A semantically correct role or `aria-live` value other than `"off"` will do this.
2. Target the "change" being programmatically determinable **at the time of the change** same techniques apply, although it kinda sorta implies immediate notification through `aria-live="assertive"` which is obviously bad.

> That may arise from an assignment of role (i.e., a role of timer means the changes are frequent and predictable). 

If we can pull off the first approach elegantly, that's great, but I'd then worry about accessibility support.  This doesn't work for the second approach because no one wants to hear a clock ticking down, or at least not at the rate of visual changes.

> My concern with your second phrase is that although it nails what we want, it is highly subjective.

That's because it was meant to nail it in plain language outside the cryptic world of success criteria ;).

> My question: to what degree can we cover concepts of importance and relevance in techniques and Understanding without making it part of the normative language?

The "importance" part directly maps to the token specified for `aria-live`, and that's the million dollar question.  The "relevance" part directly maps to `aria-relevant` and `aria-atomic`, and I'd be okay implying this by normative text with explanations in techniques and understanding.

The more I think about it, the more I think we should just narrow the scope to the stuff that always needs notification.  We're on the right track with status updates (alert, status, and progress bar roles), but we need to incorporate chat and similar logs (log role).  Then we could leave other stuff to advisory techniques and Understanding (marquee and timer roles), which are going to have questionable accessibility support anyway.

**Comment by steverep** [comment]
> Changes in content are already covered by WCAG 2.0, so I may just be misunderstanding what people are pushing for.

@awkawk, not quite true and you probably needed to be on the call and private email chain with @mbgower, @DavidMacDonald, and me to fully understand.  I promise we're working towards something clean, but still in work.  Take a look at the [scope discussion on the wiki](https://www.w3.org/WAI/GL/wiki/Live_Region_Identification#Out_of_Scope) to perhaps grasp where WCAG 2.0 is deficient.

**Comment by mbgower** [comment]
@awkawk I will attempt to summarzie the key points for you. As Steve notes, there has been a _lot_ of back and forth in email -- one of the reasons I moved it back to an issue was so that anyone could follow the discussion, at least from this point in time on.

- 4.1.2 addresses user interface components (" notification of changes to these items is available to user agents, including assistive technologies"). This new SC should address other content (which at one point we tried designating simply as 'non-user interface components') 
- a key distinction for this kind of content is that it does not take focus when content changes
- another key distinction _seems_ to be that the changes in content we are addressing are visible and largely text-based
- There are four broad considerations we need to cover:
  - indicating that a region of content has info that can dynamically update (using the term 'region' generically here)
  - indicating that a region has been updated AKA notification
  - conveying the change in content (including the removal of content-- think of  how a 'busy' status message that disappears is conveying information by its absence)
  - providing author requirements/guidance on indicating the importance and relevance of the changed content
- The [live region definition from ARIA glossary](https://www.w3.org/TR/wai-aria/terms#def_liveregion) may be starting point for definition of Change of Content 
- the SC should be complementary to 3.3.1 Error identification
- a challenge we have is that there are aria roles which designate parts of content that are dynamic BUT we can't just rely on such roles, since the attribute of aria-live can be put on any element
- Much of our vocabulary is based on ARIA, and thus not technology independent. In example, notification of content changes will frequently be supplied through the aria-live attribute
- There is a difference between the _ability_ for the user to be notified of changes in content through the UA or AT, and notifying the user of the content of the changes (the original wording allowed the latter to be an exception for the former)

- I'll also note ongoing discussion we have with Matt King about whether there should be something like aria-live="true", which designates a live region. At the moment, aria-live only designates verbosity, with aria-live="off" conveying no information. If aria-live="false" was the default boolean, then aria-live="true" would provide an ability to indicate divs, etc, that have content that can dynamically change. An attribute called something like aria-live-verbosity could provide all the current values of aria-live.

**Comment by awkawk** [comment]
@steverep ok, I'll back away, but as I do I will also say that I don't see anything on the wiki page link that you sent where the programmatic determination of the information added to the page (or changed) is not covered under WCAG 2.0. Notification of the changes is a different matter and that is where I see our gap, as well as our challenges in crisply defining an SC around.

**Comment by mraccess77** [comment]
WCAG 2 seems to address changes of context as well as changes to individual user interface components.  The area it doesn't seem to address clearly are changes/additions/removal that don't cause a change of context (don't affect a significant portion of the page) or are not specific to certain components covered by 4.1.2.

**Comment by DavidMacDonald** [comment]
Steve's plain language version
> Auto-updating and status update content can be programmatically determined, and the importance and relevance is set appropriately for assistive technologies.

This doesn't require a status update, only that if it is given, it is determinable... I like the idea of keep working in the plain language version for a bit...

> Status messages are provided for changes in content which can be programmatically determined, and the importance and relevance is set appropriately for assistive technologies.

> **Change of content:** dynamic content such as notice of success, error, progress, warning, new, changed,Auto-updating or hidden content, which result from either user action or application status changes, and which do not take focus.

**Comment by awkawk** [comment]
@mraccess77 1.3.1? e.g https://www.w3.org/TR/WCAG20-TECHS/SCR21.html

**Comment by DavidMacDonald** [comment]
Steve's plain language version
> Auto-updating and status update content can be programmatically determined, and the importance and relevance is set appropriately for assistive technologies.

This doesn't require a status update, only that if it is given, it is determinable... I like the idea of working in the plain language version for a bit...

> Status messages are provided for changes in content which can be programmatically determined, and the importance and relevance is set appropriately for assistive technologies.

> **Change of content:** dynamic content such as notice of success, error, progress, warning, new, changed,Auto-updating or hidden content, which result from either user action or application status changes, and which do not take focus. 

> **set appropriately:** cool definition here that says something like. Messages which can be read within 10 seconds at x words per minute, have the role which matches their function (marquee, status, alert, log, timer, progressbar)

**Comment by mraccess77** [comment]
@awkawk SCR21 seems more like a technique to show how to add content to a page that will be better seen by AT -- that is an accessibility supported technique -- but by itself doesn't really show any actual association or notification specifically.  In one of the examples focus is moved to the new content.

**Comment by steverep** [comment]
@DavidMacDonald,

> This doesn't require a status update, only that if it is given, it is determinable.

Right, that's by design and I don't see it as deficient.  Requiring messages is a completely different SC, and WCAG 2.0 already attempts to cover it with criteria like SC 3.3.1, Error Identification.

**Comment by awkawk** [comment]
@steverep No, I'm not going to agree that 1.3.1 is such a debacle that we should pretend that something that the WG has agreed was covered by it isn't covered in our more enlightened present view. Sorry! :)

@mraccess77 I agree that SCR21 is about adding content to a page. Once it is added to the DOM, that content is programmatically determinable, and if a user happens upon it by chance or by experience knows to look for it (e.g I passed a "Cart: 0" item when I started reading the page, I will go back to see if my item was actually added) then that user can get that information. What is missing is the the notification part, and that is what the SC we have is addressing.

**Comment by mbgower** [comment]
>  In one of the examples focus is moved to the new content.

Yes, and moving focus to new content is a technique that accomplishes notification; although where someone has the ability to move focus to new content, the content would _typically_ be a user interface component (a link in the example) and thus already be addressed in 4.1.2 .

**Comment by mbgower** [comment]
I'm thinking we may want to make our 'plain language' SC even more plain language, but before I attempt that, I want to return to what I listed as the four broad considerations we need to cover:
- indicating that a region of content has info that can dynamically update (using the term 'region' generically here)
- indicating that a region has been updated, AKA notification
-  conveying the change in content (including the removal of content-- think of how a 'busy' status message that disappears is conveying information by its absence)
-   providing author requirements/guidance on indicating the importance and relevance of the changed content

Does anyone in the discussion think this list is incorrect or over-ambitious? Is there something to add or drop?

**Comment by steverep** [comment]
@awkawk, okay okay... it's perfect.  Let me just ask the practical question then: Do you feel that marking up any of the following is covered by 1.3.1 or any other SC?

* Status updates (including errors) with ARIA `alert`, `status`, etc.
* Progress bars with ARIA `progressbar` and required attributes
* Chat or other logs with ARIA `log` etc.

I could certainly make an enlightened argument that the way those types of elements behave is information conveyed in presentation that I do not have programmatically available to me.

**Comment by steverep** [comment]
> Yes, and moving focus to new content is a technique that accomplishes notification

Please be very careful touting that as a technique - in most cases it is inappropriate and would violate [SC 3.2.5, Change on Request](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-no-extreme-changes-context.html).  For example, if I click "add to cart" and get tossed way up the page to the "cart with X items" link, that's no good.

**Comment by awkawk** [comment]
> - Status updates (including errors) with ARIA alert, status, etc.
> - Progress bars with ARIA progressbar and required attributes
> - Chat or other logs with ARIA log etc.

@steve, I'm not sure what you are saying isn't programmatically determinable. If you have an alert, that needs a role to be conveyed per 4.1.2. I'm not sure what does or doesn't work for you with alerts, but I don't think that you are saying that the content of the alert isn't programmatically determinable, are you?

Similarly for the other items - ARIA roles exist to help provide roles and semantics to otherwise anonymous content, but the content itself is readable on the web page so long as it is presented there, right?

Again, I'm trying to back away while you all figure out the latest proposal, but I do hope that it isn't repeating existing requirements.

**Comment by DavidMacDonald** [comment]
@awkawk 

There is no requirement in the WCAG to provide a status message for changed content if the user is not focused on it. This is a user requirement, and many screen reader users are asking for this.

**Comment by steverep** [comment]
@awkawk,

> I'm not sure what you are saying isn't programmatically determinable. If you have an alert, that needs a role to be conveyed per 4.1.2. I'm not sure what does or doesn't work for you with alerts, but I don't think that you are saying that the content of the alert isn't programmatically determinable, are you?

No, I (and at least Mike and David) are saying that your assertion that an alert, such as an error message after a form submission, is covered by 4.1.2 is not correct.  That SC covers "user interface components" only, which WCAG scopes as controls, and none of those 3 examples I mentioned are controls.  In the case of a status or alert message, it's usually just text.

**Comment by DavidMacDonald** [comment]
I should also mention that the previous SC language has had broad support, whenever I've presented it at conferences and talks. Although I acknowledge it had a few risks of allowing too much chattiness.

**Comment by awkawk** [comment]
@steverep I need you to be more specific. When you say "[my] assertion that an alert... is covered by 4.1.2 is not correct" - what specifically are you saying isn't covered, either by 4.1.2 or another 2.0 SC?

@DavidMacDonald I never suggested (at least not deliberately) that WCAG 2.0 had a requirement to provide a status message for changed content. I was only reacting to the suggested text that the status message itself needed to be programmatically determinable. If it appears on screen then it needs to be programmatically determinable, per 1.3.1.

**Comment by mraccess77** [comment]
At the very least this shows the lack of clarity for experts on whether WCAG 2.0 does or does not require markup communicating dynamic regions of a page.  Once consensus is reached we should clarify this in the WCAG 2.0 documents.

**Comment by DavidMacDonald** [comment]
@awkawk 
Right. 

The intent of this SC is to ensure Screen Reader users are notified of changes that happen on the page that are not under their focus, in a way that doesn't interrupt their work.

**Comment by mbgower** [comment]
Up half the night thinking about this one. I'm going to get back to my 4 bullets and respond to a couple of questions, and offer some comments and thoughts I have had. 

1.  indicating that a region of content has info that can dynamically update (using the term 'region' generically here)
1.  indicating that a region has been updated, AKA notification
1.  conveying the change in content (including the removal of content-- think of how a 'busy' status message that disappears is conveying information by its absence)
1.  providing author requirements/guidance on indicating the importance and relevance of the changed content

### 1. indicating content that can dynamically update

The very first line of the [aria-live property](https://www.w3.org/TR/wai-aria/#aria-live) states
> Indicates that an element will be updated, and describes the types of updates the user agents, assistive technologies, and user can expect from the live region.

then the characteristic of "off" is described as
> Indicates that updates to the region should not be presented to the user unless the used is currently focused on that region.

I've asked for clarification from Matt King on this, but it seems to me that between the live regions and the aria-live attribute, we do have this ability to _indicate_.

### 2. Notifiy that an update has occurred
Built into the live region roles and aria-live property.

### 3. Convey the contents/nature of the change
For all but content that is completely removed, handled by live regions and aria-live property. We'll need to provide guidance in what to do to indicate that content has been removed. As one critical example, when the "Busy" status indicator disappears, screen readers can definitely benefit from knowing that.

### Number 4, importance and relevance of changed content.
@DavidMacDonald said
> I had that in the Understanding, talking about short messages. But if possible this should be in the SC, ensuring the page isn't too chatty.

My position, after much thought, is that this should actually be able to be covered by techniques that are also included in 1.3.1. Why? Because the importance and relevance for information added to the presentation should be based on how it is presented. Matt King responded "Bingo" to this statement in our email thread
> Since the new text is in the UI, there tends to be a fair degree of consideration placed on the brevity and effectiveness of the information. I'd argue pretty strongly that making such information available through live regions is the most elegant and reliable way of ensuring a screen-reader user gets a similar experience. I'd also argue that in the same way location helps a designer decide on appropriate regions, the placement and frequency of information could help the designer designate the relative importance of that information to everyone, including a screen reader user.

A lot of this guidance is built into the default values of the live region roles, and the [APG](https://w3c.github.io/aria-practices/). I think some techniques on using aria-live appropriately based on the relative position of changed content in relationship to the item with focus, can resolve this conundrum, without us having to put any language into the SC itself.

---
### Conclusion
I think we can cover ourselves with a shorter SC, that cross-references 4.1.2 and 1.3.1 in Notes. Two candidates that seem to be approaching this:

>Changes in content are programmatically determinable for auto-updating content and status information

OR
> Auto-updating and status information can be programmatically determined.

Steve had [some thoughts](https://github.com/w3c/wcag21/issues/544#issuecomment-341185804) on variations. Both need some variation on the following definition:
> **status information**
  dynamic content such as success, progress or warning notices, which result from either user action or application status changes, and which do not take focus

I'd also suggest we want to call out something like.
Note: See 4.1.2 Name, Role, Value for considerations involving changes to the content of user interface components.

**Comment by mbgower** [comment]
By the way, another thought I had last night. What about prompts, as in speech interaction? I believe that conversational UI design is going to have to be brought into this as a consideration. 

It _looks_ like we have a gap for speech conversations, period. It's not covered by any of the time-based media, since the one most relevant, [Audio-only (live)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-live-audio-only.html) is AAA and [Captions (live)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-real-time-captions.html) is restricted to synchronized media.

I think a good portion can potentially get covered under 1.1.1 (although "presentation" is never defined). [IBM added our own Situation G to our techniques](http://www-03.ibm.com/able/guidelines/ci162/non_text_content.html) to cover audio alerts. I flagged that to the WG previously. I see a potential for an audio situation getting added that covered both audio alerts and speech prompts. That would ensure a text equivalent for audio alerts, which would then potentially get covered by existing WCAG considerations (although as an alternative, would they get the same pass that captions do for some things?).

It might be an idea to consider speech prompts here to see if we can bring them in. It would be pretty hard to argue they fall under 'status information' or 'auto-updating'.

**Comment by mbgower** [comment]
@awkawk  said
> I need you to be more specific. When you say "[my] assertion that an alert... is covered by 4.1.2 is not correct" - what specifically are you saying isn't covered, either by 4.1.2 or another 2.0 SC?

The opening of 4.1.2 makes it clear it is "For all User Interface Components". The glossary definition:
>user interface component
    a part of the content that is perceived by users as a single control for a distinct function
    Note 1: Multiple user interface components may be implemented as a single programmatic element. Components here is not tied to programming techniques, but rather to what the user perceives as separate controls.
    Note 2: User interface components include form elements and links as well as components generated by scripts.
    Example: An applet has a "control" that can be used to move through content by line or page or random access. Since each of these would need to have a name and be settable independently, they would each be a "user interface component." 

Information we are concerned about has many qualities which clearly separate it from this definition:

- It tends to be text only (not a widget)
- It is non-interactive (cannot be navigated to by keyboard without use of AT).
- It does not normally take focus
  - which follows that it is updated when focus is elsewhere

4.1.2 covers many considerations to do with such components, such as assignment of role, changes in state, properties and values, and notification of changes.

We are trying to ensure that some of these considerations are addressed for status text and other kinds of information which does not meet the definition of a user interface component. At one point we thought we could cover the topic by simply saying such items need to be assigned an appropriate live role. However, there are a couple of issues with this. First, it is technology depenedent. Second, even within aria specs, aria-live is a property that can be placed on any element. That implies there are likely cases where insisting on use of a live region will be problematic.

In a nutshell, we have one solution for the problem. We just need to properly describe the problem with SC language so that we can ensure the solution can be properly applied without being misapplied/misinterpreted.

**Comment by DavidMacDonald** [comment]
@mbgower 
I can live with some form of this.

1. Should status messages be visual to cover off part of Issue #54, and to address your concern about internationalization?
2. I know "warnings" could include "errors", but I'd put errors in that definition.
3. regarding audio feedback. I'm OK with that idea, but COGA may be concerned if all audio feedback has to get a text alternative, because it might be a disincentive to authors who might otherwise do something that helps them... let's mull that over at TPAC

**Comment by mbgower** [comment]
Okay, thanks for the response @DavidMacDonald . I also saw the email feedback from Matt and you on the current wording of the aria spec for aria-live. Still digesting that.

I have looked at @steverep 's and @awkawk's prior feedback and attempted to work this into my latest language. I'm still not happy with this, but _hoping_ it may actually approach something we can all live with:

---

### 3.2.7 Change of Content
For auto-updating or status information, all of the following are true:
 1.  Changes in content are programmatically determinable through role or attributes
 2.  Notification of changes to content is available to user agents, including assistive technologies.

**status information**
content such as success, progress or error notices, which dynamically updates as a result of user action or application status changes, and which do not take focus

---

 I added in the idea of roles in the first item. Originally, it just read:
> 1. Changes in content are programmatically determinable 

Comments? My idea would be to keep it simple, and that a lot of the nuance would be clarified through Understanding and techniques.

David, I think I'm going to leave audio feedback out of this for now. Once we have language that we can agree on for the main thing, we can try to incorporate. But I think trying to work it in before that just makes it messier.

**Comment by mraccess77** [comment]
The phrase "auto-updating and status information" is too restrictive in my opinion.  It doesn't seem to capture the issue when new content appears and the presence of the new content not only needs to be programmatically but also a notification given when the new content appears.

**Comment by mbgower** [comment]
Hi @mraccess77, can you give me examples of what content you feel is missing? The definition of status information reads as follows

>status information
content such as success, progress or error notices, which dynamically updates as a result of user action or application status changes, and which does not take focus

As you are aware, Auto-updating is listed in 2.2.2. It is not defined there, but qualified to the degree it appplies in that content. We haven't discussed whether there is a need to do more than cross-reference it. Obviously the intent would be to provide Understanding and technique guidance for specific types of auto-updating content.

>Auto-updating: For any auto-updating information that (1) starts automatically and (2) is presented in parallel with other content, there is a mechanism for the user to pause, stop, or hide it or to control the frequency of the update unless the auto-updating is part of an activity where it is essential.

**Comment by mbgower** [comment]
>  It doesn't seem to capture the issue when new content appears and the presence of the new content not only needs to be programmatically but also a notification given when the new content appears.

@mraccess77, the two bullets of the SC seems to directly address this. The first has the requirement for the changes to be determinable. The second requires notification of those changes. Can you elaborate on how this is deficient?

**Comment by mbgower** [comment]
In discussion with Steve, we're wondering about removing the auto-updating information to steer clear of confusion over use of less adopted and supported aria roles (These could be added in as advisory techniques.) and concentrate on status messages and error logs.  As a first shot at that, I'm simply removing the term

### 3.2.7 Change of Content
For status information, all of the following are true:
- Changes in content are programmatically determinable through role or attributes
- Notification of changes to content is available to user agents, including assistive technologies.

**status information**
content, such as success, progress or error notices, which dynamically updates as a result of user action or application status changes, and which does not take focus

-- 
We could also potentially bake the definition into the SC wording now:
For content which does not take focus and which dynamically updates as a result of user action or application status changes, the following are true:
...
Note: success, progress and error notices are examples of such content

**Comment by mraccess77** [comment]
Focusing on status messages and error logs doesn't really seem to address the issues with new content.  I am thinking of a situation where you check a box and a section above appears with new form elements in it.  This is an important content change that isn't a change of context but it is something the user needs to be aware of.  Are we putting off this scenario for WCAG 2.1 or trying to address this situation?

**Comment by mbgower** [comment]
@mraccess77 said
>  Focusing on status messages and error logs doesn't really seem to address the issues with new content. I am thinking of a situation where you check a box and a section above appears with new form elements in it. 

I would think of that as a Change of Context and failure of On Input.
Here is the wording in [SCR19 Using an onchange event on a select element without causing a change of context](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/SCR19)
>It is important to note that the select item which is modified is after the trigger select element in the reading order of the Web page. This ensures that assistive technologies will pick up the change and users will encounter the new data when the modified element receives focus. This technique relies on JavaScript support in the user agent. 

Here are key parts of the [glossary definition of changes of context](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html#context-changedef):
>major changes in the content of the Web page that, if made without user awareness, can disorient users who are not able to view the entire page simultaneously
Changes in context include changes of...content that changes the meaning of the Web page.

You state
>This is an important content change that isn't a change of context but it is something the user needs to be aware of. 

As you note, this is an important content change _and_ the user needs to be aware of it. This seems to meet the criterion of the fourth bullet in the definition, as quoted, and I have had no problem failing such instances with current 2.0 pages.

---

On a more general note, you are correct in that we are not trying to resolve every situation where a change of content could be conveyed by the app to the AT. We are trying to tackle specific ones which are least detectable to a screen reader user (status messages, as defined, would never take focus), and for which there are good existing ARIA roles and guidance. Low hanging fruit as it were.

I think we can also list considerations in the Understanding document that talk about considerations where new content does not meet our subset, and provide advisory techniques for those situations (everything from moving focus to new content that can take focus, to use of advanced notice, dialogues and offscreen text).

**Comment by mbgower** [comment]
Cross referencing[ a comment in another issue](https://github.com/w3c/wcag21/issues/539#issuecomment-343269743) that is relevant to discussion

**Comment by mraccess77** [comment]
@mbgower it's not at all clear that it's a change of context.  In my example a section appeared -- but what if just one new field appears and the triggering element was a button that the user pressed enter on.  In that case it was user initiated and one new field would not seem to be a change of context.

**Comment by mbgower** [comment]
>what if just one new field appears and the triggering element was a button that the user pressed enter on

If the element appeared based on such a user action, I agree that it is not a failure in 2.0 against either 3.2.1 On Input or 3.2.2 On Focus, given that it did not occur when an element received focus or the user began inputting information.

I'm going to address your scenario, but I do think it is getting off base of the catalyst for this SC -- that is, to surface text added to the screen which doesn't (read "cannot") take focus. Given that we've already acknowledge this SC is aimed at targetting the most problematic and easily solved situation, and is not designed to solve every situation where new content is added to the screen, I'm not sure how germane the digression will be.

---
First off, I think we can all agree that any interactive control that is designed to appear above the current focus point is going to be a challenge to discover for many users, and is going to be confusing/surprising to anyone who renavigates the site and discovers it. So, we're talking about poor design. We cannot solve every crappy design quirk with standards.

Second, I think we can also agree that this situation becomes worse for people with a variety of disabilities. 

### What are the ways to address the scenario that you describe?
1. I can convince the team that this is a crappy design and convince them to make their UI better. If they've followed any kind of UAT process, they'll have received feedback that supports this. I can also point out the increased problems for PwDs, and even take a stab at saying how it is a failure (probably using [2.4.3 Focus Order](
https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html) or [Conformance Requirement 4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/conformance.html#uc-conf-req4-head)  ). This is obviously the preferred solution.

1. I can put the focus to this new control, which would certain make its presence clear to everyone. Given this was triggered by a user activating a button, such a change of context is allowed. I can't imagine that's a very elegant solution, but it does solve an inelegant design problem.

1. I could use some kind of onscreen alert to let users know a new field has been added above the present location. I'm hard-pressed to think this is desirable, but it does let all user know to go back. Such an alert would either be modal and announced, or would be covered by this new SC and its techniques, which cover alert roles and aria-alert.

1. I could only worry about alerting blind users; essentially giving the same info as in the prior scenario, but do it with non-displaying text.

1. I could have the new element's contents (name, role, value) announced by the screen reader without moving focus. Probably the least desirable of these options, given I would have no context for what was spoken or why.

1. I could combine the last two and both announce a new field was added and read out its label/value. Slightly better? Only know with user testing.

---
All these solutions could be done with current language. Note that these last 3 are all solutions restricted to users with screen readers.  What we'd be ensuring with the new SC language is that in scenario three where the visible notice appears as the solution for all users, the screen reader user also receives that notice.

**Comment by mbgower** [comment]
Suggested wording change on Monday's working group call was:
> For content which does not take focus and which dynamically updates as a result of user action or application status changes, notification of changes to content can be programmatically determined by user agents, including assistive technologies.

**Comment by mbgower** [comment]
Comments from today's discussion:
- Steve wanted to remove "user action"
- James questioned whether this ruled out moving focus
- he also asked about content that appears at the same time as content which takes focus which _doesn't_ take focus, such as a bunch of tabular data.

**Comment by mbgower** [comment]
@steverep  and @DavidMacDonald .Attempt at a rewrite to take these into account:

> For status messages which do not take focus and which dynamically update, notification of changes of content can be programmatically determined by user agents, including assistive technologies.

>**Status messages:**
Notices that provide status information to the user, such as the success of an action, the waiting state of an application, the progress of a process or the existence of errors.

I also wanted to note that the "including assistive technologies" phrase was listed directly from 4.1.2. I assume there was a reason it was included there, so it makes sense to reuse here.

**Comment by mbgower** [comment]
In discussion, David raised the idea of ensuring we capture a "18 results" message on search results as an example of a status message

**Comment by mbgower** [comment]
David wants to distinguish between notifications of new content and all that new content being read out. The former is a status notices; the latter is chatter no one wants

**Comment by mbgower** [comment]
Final wording:
> Status messages are programmatically determinable through role or attributes such that they can be announced by assistive technologies without receiving focus.

**Status messages**
Notices that provide information to the user, such as the success of an action, the waiting state of an application, the progress of a process or the existence of errors.

Status content is an alternative? 
Fine with throwing in "In markup languages..."

**Comment by jasonjgw** [comment]
I suggest "can be programmatically determined" and "can be presented to the user" (rather than "announced", which suggests a spoken interface only). These changes are editorial.

**Comment by DavidMacDonald** [comment]
Incorporating Jason's comments:

> Status messages are programmatically determined through role or attributes such that they can be presented to the user by assistive technologies without receiving focus.

**Comment by mbgower** [comment]
Yes, I think Jason's change addresses the comments Steve had at the end which we were going to tackle by including the following in the Understanding: “announced” can cover a variety of modalities for surfacing information to the user from sound to spoken text to Braille display.

Nice to have someone look at it fresh after we're all punch drunk with fatigue :)

The final things Steve had requested were some revisions to the definition of status messages -- David had a few words about that too -- and a consideration to renaming the SC.
First objective: remove the 'such as qualifier' in opposition, which arguably meant that any information provided to the user could be a notice:
>Notices that provide information to the user on the success of an action, the waiting state of an application, the progress of a process, the existence of errors, or notification of changes to content.

Objective two: consider working in the idea of this being dynamic content that is not present in the DOM initially. I also worked in David's desire to track changes to the UI through user action by adding in "or results of"
> Changes in content that provide information to the user on the success or results of an action, the waiting state of an application, the progress of a process, the existence of errors, or similar notifications.

I'm pretty happy with that. 

Objective three: consider a new name for the SC. I guess the obvious title is Status Messages. But I feel like we're pushing our luck a bit with this. My feeling is that if we leave it as Change of Content it will give us some leeway in the Understanding document to riff on some of David's considerations. I also suspect that someone in the working group will propose a new title, and it may be prudent to give them that bone. In some ways I think the WG does not want to accept something without changing it. Maybe if we put the focus to the group by asking that question on the title, we can reduce the likelihood of rehashing the whole thought processes that lead us to this proposed wording in the first place :)

The submitted version would read:
>**3.2.7 Change of Content**
>Status messages are programmatically determined through role or attributes such that they can be presented to the user by assistive technologies without receiving focus.

>**Status messages**
Changes in content that provide information to the user on the success or results of an action, the waiting state of an application, the progress of a process, the existence of errors, or similar notifications.

**Comment by mbgower** [comment]
Final thought. Do we need to define status messages as NOT being user interface components? I'm worried that someone might make the argument that if the status message is in a dialog or some other form of notice that takes focus, this SC seems redundant and possibly even disruptive.

Changes in content that are not changes of context, which provide...

**Comment by steverep** [comment]
@mbgower, I like the latest definition which starts by saying "changes in content..."  That solves one issue.

I don't think we should leave the title as is; it should reflect the scope clearly.  Something like "Status Messages" or "Status Changes" would work best.

And I think that the SC saying without focus is enough.  As I said last night, we don't want an "out" just by putting `tabindex=0`.  The behavior of focus is what matters, not whether it can or cannot have focus.

**Comment by mbgower** [comment]
@DavidMacDonald @steverep @jasonjgw, please offer final comments:

**3.2.7 Change of Content**
Status messages are programmatically determined through role or attributes such that they can be presented to the user by assistive technologies without receiving focus.

**Status messages**
Changes in content that provide information to the user on the success or results of an action, the waiting state of an application, the progress of a process, the existence of errors, or similar notifications.
OR
**Status messages**
Changes in content that are not changes in context, that provide information to the user on the success or results of an action, the waiting state of an application, the progress of a process, the existence of errors, or similar notifications.

**Comment by steverep** [comment]
1. I still think we need to change the name to reflect the scope.
2. Change "attribute" to "properties" because it's less tech-specific, `role` is an attribute, and 4.1.2 uses "properties.
3. I don't see any need to confuse the definition by mentioning changes in context.  If I'm catching on to what you're worried about, then just move the "without taking focus" part into the definition.  For example, by saying "status messages do not take focus when they appear or change."

**Comment by mbgower** [comment]
Oops, sorry about missing new title!
The issue with putting the 'without taking focus' into the defn is that then someone could say 'i've met it. just move the cursor to the new text and jaws reads it'. I'm okay with crossing that bridge when we have to and leave without the context qualifier -- although I think that works!

**3.2.7 Change of Content**
Status messages are programmatically determined through role or attributes such that they can be presented to the user by assistive technologies without receiving focus.

**Status changes**
Changes in content that provide information to the user on the success or results of an action, the waiting state of an application, the progress of a process, the existence of errors, or similar notifications.

**Comment by spanchang** [comment]
This SC is very different from all other SCs as it  specifically dictates how notifications are to be made programmatically determinable: i.e. using role or other attribute. Is that not how everything is made accessible? By using a particular tag or attribute? 
So why specify this requirement for this SC? 
By the way it is documented that relying on UA/AT support for accessibility-markup is ok for satisfying some SCs like SC 2.4.4 (reading linked context). That is a situation where one can read context without moving focus from the link.

I am glad to see  that the originally proposed SC for "change in content" has undergone a significant metamorphosis since I raised the objection in issue 539.
Apparently there is  consensus on the objections raised in the articles:
http://www.mindoversight.com/?p=171
and
WHAT’S THE VOID IN WCAG 2.0 THAT THE PROPOSED "CHANGE IN CONTENT" SUCCESS CRITERION 3.2.6 ATTEMPTS TO BRIDGE?
(http://www.mindoversight.com/?p=168)
Thanks and best wishes,
Sailesh

**Comment by spanchang** [comment]
Will a JS alert satisfy the requirement?

**Comment by steverep** [comment]
@spanchang, this transformation is now all about simply using the ARIA live region roles and their properties appropriately.  The ones targeted here are those which are well accessibility supported: alert, status, log, and progressbar.  Results can also be achieved by specifying `aria-live` and the other 3 properties separately or in conjunction with the role, but often the role defaults will suffice.

A JS alert is not in scope here because a screen reader will jump focus to that.  In other words, it does not meet the definition of a status message per this SC.

**Comment by mbgower** [comment]
> Will a JS alert satisfy the requirement?

To build on what Steve is saying, any info displayed through a js alert triggers a change in context, and so, according to the definition of status message is not applicable. 

Language for status message definition agreed on today's call:
> Changes in content that are not changes in context, that provide information to the user on the success or results of an action, on the waiting state of an application, on the progress of a process, or on the existence of errors. 

There's nothing wrong with using a dialog to alert users of issues. This SC is covering a situation where an author  uses javascript to add text (i.e., whatever the author put into the alert in your example) directly into the page content instead. It doesn't take focus, and there's no way for the AT to understand it has been added unless it is given the proper role AND any properties that role has that cover communication to the AT are appropriately set for notice (i.e., it cannot put aria-live="off") and alert the AT.

**Comment by mbgower** [comment]
> Apparently there is consensus on the objections raised in the articles:
http://www.mindoversight.com/?p=171

I wouldn't say there is consensus, but there was a decision made to reduce the scope of the SC down to something that could be adopted successfully in the working group's timeline, which would still be of demonstrable benefit.

**Comment by spanchang** [comment]
Mike / Steve,
If text / content is added that is not visually distinguishable  from
existing content on the page, yes adding role / other properties will
help AT users.
But if that content lacks distinct presentational properties, will
that content not be difficult for some sighted non-AT using PWD user
groups to notice? Some of these users could be those whose needs are
being addressed by Cognitive and Learning Disabilities Accessibility
Task Force.
It could be a UI design / usability issue too that results in ordinary
users too missing the changed content unless they scrutinize the page
carefully.
This situation is discussed under Reason#2 of
http://www.mindoversight.com/?p=168
If presentational / styling properties are added, it will then be
covered by SC 1.3.1 because that content has to be programmatically
determinable too.

Or at the present time, is it a conscious decision to not consider the
needs of non-AT using PWD groups?
I believe this is the case, right?
Thanks and best wishes,
Sailesh


On 11/30/17, Mike Gower <notifications@github.com> wrote:
>> Apparently there is consensus on the objections raised in the articles:
> http://www.mindoversight.com/?p=171
>
> I wouldn't say there is consensus, but there was a decision made to reduce
> the scope of the SC down to something that could be adopted successfully in
> the working group's timeline, which would still be of demonstrable benefit.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/w3c/wcag21/issues/544#issuecomment-348315950


-- 
Sailesh Panchang
Principal Accessibility Consultant
Deque Systems Inc
Phone 703-225-0380 ext 105
Mobile: 571-344-1765

**Comment by mbgower** [comment]
@spanchang 
> But if that content lacks distinct presentational properties, will that content not be difficult for some sighted non-AT using PWD user groups to notice?

Possibly. But what presentational properties do you think would ensure all non-AT using PwDs would notice it? That is in the realm of conjecture. And if you could define any, would you want to insist those must be used by all designers? What we can say is that if such text is given an appropriate role to indicate it is a status message or whatever, that is arguably beneficial to any AT which wishes to do something with that, for whatever user group.

> It could be a UI design / usability issue too that results in ordinary users too missing the changed content

Sure, but the task of the guidelines is not to prevent designers from making poor designs. It's to try to offset poor design decisions that adversely affect PwDs to a larger degree.

> If presentational / styling properties are added, it will then be covered by SC 1.3.1 because that content has to be programmatically determinable too.

I think you're missing the point here. While something that is visually presented as larger text can be given a heading tag -- which will meet 1.3.1 -- that does nothing to alert anyone to the fact it has been added to the DOM after page load.

If an author brings such information into the UI as any kind of dialog, then it is handled fine by existing 2.0 requirements. But there are a growing list of situations where the author does not change the context of the user by pulling way focus in that manner; the author is content with assuming a user will see the new content being added. 
Adding new content to the screen can happen from any number of motivations. We cannot conjecture on what those are. What we have done is attempted to define text updates that have the purpose of providing status information to the user. These are the most crucial for any user to detect in near real time. 

> is it a conscious decision to not consider the needs of non-AT using PWD groups?

I'm not happy with the way the wording got changed from 'user agents including assistive technology' to just assistive technologies. But that was an edit requested during review of the SC language. I'd prefer that get put back in, as I agree there is not clear reason to not specify user agents. That said, if the info is available to ATs, it is, I believe, by definition available to user agents since they are the ones handling the API. So I'm not sure it doesn't amount to the same thing.

**Comment by spanchang** [comment]
Hello Mike,
Thanks for the explanation.
No I am not missing the point ... if it is  injected into the page and
in a manner that is visually distinguishable it does serve as an alert
or status update for the user and then needs the role of alert or such
assigned to it. The heading markup will help one discover that content
if one needs to  do so separately.
I am aware WCAG is there to level poor design choices so content is
accessible. That's why I asked about sighted non-AT using PWD groups.
Also, if the intent is to make the new content perceivable, should
this SC  not be under Guideline 1.3 then? And at Level A?
Best regards,
Sailesh


On 12/5/17, Mike Gower <notifications@github.com> wrote:
> @spanchang
>> But if that content lacks distinct presentational properties, will that
>> content not be difficult for some sighted non-AT using PWD user groups to
>> notice?
>
> Possibly. But what presentational properties do you think would ensure all
> non-AT using PwDs would notice it? That is in the realm of conjecture. And
> if you could define any, would you want to insist those must be used by all
> designers? What we can say is that if such text is given an appropriate role
> to indicate it is a status message or whatever, that is arguably beneficial
> to any AT which wishes to do something with that, for whatever user group.
>
>> It could be a UI design / usability issue too that results in ordinary
>> users too missing the changed content
>
> Sure, but the task of the guidelines is not to prevent designers from making
> poor designs. It's to try to offset poor design decisions that adversely
> affect PwDs to a larger degree.
>
>> If presentational / styling properties are added, it will then be covered
>> by SC 1.3.1 because that content has to be programmatically determinable
>> too.
>
> I think you're missing the point here. While something that is visually
> presented as larger text can be given a heading tag -- which will meet 1.3.1
> -- that does nothing to alert anyone to the fact it has been added to the
> DOM after page load.
>
> If an author brings such information into the UI as any kind of dialog, then
> it is handled fine by existing 2.0 requirements. But there are a growing
> list of situations where the author does not change the context of the user
> by pulling way focus in that manner; the author is content with assuming a
> user will see the new content being added.
> Adding new content to the screen can happen from any number of motivations.
> We cannot conjecture on what those are. What we have done is attempted to
> define text updates that have the purpose of providing status information to
> the user. These are the most crucial for any user to detect in near real
> time.
>
>> is it a conscious decision to not consider the needs of non-AT using PWD
>> groups?
>
> I'm not happy with the way the wording got changed from 'user agents
> including assistive technology' to just assistive technologies. But that was
> an edit requested during review of the SC language. I'd prefer that get put
> back in, as I agree there is not clear reason to not specify user agents.
> That said, if the info is available to ATs, it is, I believe, by definition
> available to user agents since they are the ones handling the API. So I'm
> not sure it doesn't amount to the same thing.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/w3c/wcag21/issues/544#issuecomment-349492880


-- 
Sailesh Panchang
Principal Accessibility Consultant
Deque Systems Inc
Phone 703-225-0380 ext 105
Mobile: 571-344-1765

**Comment by mbgower** [comment]
>  if it is injected into the page and in a manner that is visually distinguishable it does serve as an alert or status update for the user and then needs the role of alert or such assigned to it.

Ah, I think I understand what you're saying: you want the status role assigned _and_ a requirement for some kind of programmatic indicator through an attribute or whatever that makes the new content meet 1.3.1. Is that it?

If that is what you're asking for, that requirement already exists _if_ there is any kind of styling applied to the text. If there's no styling, I agree there is nothing on which to hang an accessibility requirement for 1.3.1.

**Comment by spanchang** [comment]
Mike, You wrote:
<start>
Ah, I think I understand what you're saying: you want the status role assigned and a requirement for some kind of programmatic indicator through an attribute or whatever that makes the new content meet 1.3.1. Is that it?
If that is what you're asking for, that requirement already exists if there is any kind of styling applied to the text.
</end>
Consider the example at http://mars.dequecloud.com/demo/GlobalRedAlert.htm
Submit the form unfilled / partly filled and notice the error added within the h3 above the form. 
Your statement above suggests you agree that failing to mark up the span tag within the h3 above the form for the global error message injected to the page fails SC 1.3.1 today, correct?
If so, should the status change SC not exclude such cases?
Ref Dec 7 2017 draft: https://www.w3.org/TR/WCAG21/
<start>If there's no styling, I agree there is nothing on which to hang an accessibility requirement for 1.3.1.</end>
In this case, a new SC under Guideline 1.4 should require that such content is distinguishable via its presentation properties. 
Thanks,
Sailesh

**Comment by mbgower** [comment]
> Your statement above suggests you agree that failing to mark up the span tag within the h3 above the form for the global error message injected to the page fails SC 1.3.1 today, correct

If the message appears with the same formatting you use on your site for headings, then if the author does not mark it up with a heading tag, it seems like a pretty clear failure of 1.3.1.  But as I hope is by now clear, that has little to do with the intent or wording of the Status Changes SC.

> If so, should the status change SC not exclude such cases?

Why would failing to put your span inside the H3 have any bearing on whether the new text is surfaced to ATs? I feel like we're speaking at cross purposes here. There are no styling tags in HTML that are going to accomplish providing a 'notice' to users. Where you implement your text in something that causes a change of context, you accomplish that automatically, which is why that use case is excluded.

> If there's no styling, I agree there is nothing on which to hang an accessibility requirement for 1.3.1.
In this case, a new SC under Guideline 1.4 should require that such content is distinguishable via its presentation properties.

You are welcome to propose such an SC, but that is an independent issue from what is being tackled here. The H3 in your example does not increase the ability for an AT to surface the new information -- that is accomplished with your proper use of an aria-live attribute. This SC ensures that in a situation like this, the aria-live attribute or an equivalent method of flagging the text is implemented.

@spanchang now that the SC has reached CFC and is part of the latest draft, I suggest it makes sense for you to open specific issues for the SC so that they can be considered by the larger working group.

**Comment by spanchang** [comment]
Hi Mike,
<start>Why would failing to put your span inside the H3 have any
bearing on whether the new text is surfaced to ATs? I feel like we're
speaking at cross purposes here. There are no styling tags in HTML
that are going to accomplish providing a 'notice' to users. Where you
implement your text in something that causes a change of context, you
accomplish that automatically, which is why that use case is excluded
... The H3 in your example does not increase the ability for an AT to
surface the new information </end>
I think you are missing the point in this example. The h3 is not the
essence of the example. The span tag within the h3 is styled
differently so it is visually  discernible and it has been assigned
aria-live=assertive (i.e. like an alert)  so it is instantly detected
by screen readers when presented. In this example it is within the h3
but it could be a separate p-tag too for instance. So the
info-relation now in that example is programmatically determinable
because the span tag has proper semantics / markup.
You said "that requirement already exists if there is any kind of
styling applied to the text", so the example tries to illustrate that.
Sailesh
On 12/15/17, Mike Gower <notifications@github.com> wrote:
>> Your statement above suggests you agree that failing to mark up the span
>> tag within the h3 above the form for the global error message injected to
>> the page fails SC 1.3.1 today, correct
>
> If the message appears with the same formatting you use on your site for
> headings, then if the author does not mark it up with a heading tag, it
> seems like a pretty clear failure of 1.3.1.  But as I hope is by now clear,
> that has little to do with the intent or wording of the Status Changes SC.
>
>> If so, should the status change SC not exclude such cases?
>
> Why would failing to put your span inside the H3 have any bearing on whether
> the new text is surfaced to ATs? I feel like we're speaking at cross
> purposes here. There are no styling tags in HTML that are going to
> accomplish providing a 'notice' to users. Where you implement your text in
> something that causes a change of context, you accomplish that
> automatically, which is why that use case is excluded.
>
>> If there's no styling, I agree there is nothing on which to hang an
>> accessibility requirement for 1.3.1.
> In this case, a new SC under Guideline 1.4 should require that such content
> is distinguishable via its presentation properties.
>
> You are welcome to propose such an SC, but that is an independent issue from
> what is being tackled here. The H3 in your example does not increase the
> ability for an AT to surface the new information -- that is accomplished
> with your proper use of an aria-live attribute. This SC ensures that in a
> situation like this, the aria-live attribute or an equivalent method of
> flagging the text is implemented.
>
> @spanchang now that the SC has reached CFC and is part of the latest draft,
> I suggest it makes sense for you to open specific issues for the SC so that
> they can be considered by the larger working group.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/w3c/wcag21/issues/544#issuecomment-352042641


-- 
Sailesh Panchang
Principal Accessibility Consultant
Deque Systems Inc
Phone 703-225-0380 ext 105
Mobile: 571-344-1765

**Comment by steverep** [comment]
I think there is an outstanding task for this SC to be moved to Guideline 1.3, Adaptable.  Its current home under 3.2, Predictable, seems very much incorrect.  Anyone disagree?

**Comment by mbgower** [comment]
Closing issue, since SC has been added to December published working draft.
